src/public.rs

```diff
@@ -178,58 +159,33 @@ mod use_std {
             unsafe { Self::is_slice_in_curve_unchecked(self.as_ref()) }
         }
 
-        /// Recover the public key from a signature performed with
-        /// [`SecretKey::sign`]
-        pub fn recover<M>(signature: Signature, message: M) -> Result<PublicKey, Error>
-        where
-            M: AsRef<[u8]>,
-        {
-            let message = SecretKey::normalize_message(message);
+        pub(crate) fn from_secp(pk: &Secp256k1PublicKey) -> PublicKey {
+            debug_assert_eq!(
+                UNCOMPRESSED_PUBLIC_KEY_SIZE,
+                secp256k1::constants::UNCOMPRESSED_PUBLIC_KEY_SIZE
+            );
 
-            Self::_recover(signature, &message)
-        }
+            let pk = pk.serialize_uncompressed();
 
-        /// Recover the public key from a signature performed with
-        /// [`SecretKey::sign`]
-        ///
-        /// # Safety
-        ///
-        /// The protocol expects the message to be the result of a hash -
-        /// otherwise, its verification is malleable. The output of the
-        /// hash must be 32 bytes.
-        ///
-        /// The unsafe directive of this function is related only to the message
-        /// input. It might fail if the signature is inconsistent.
-        pub unsafe fn recover_unchecked<M>(
-            signature: Signature,
-            message: M,
-        ) -> Result<PublicKey, Error>
-        where
-            M: AsRef<[u8]>,
-        {
-            let message = SecretKey::cast_message(message.as_ref());
-
-            Self::_recover(signature, message)
-        }
+            debug_assert_eq!(SECP_UNCOMPRESSED_FLAG, pk[0]);
+
+            // Ignore the first byte of the compressed flag
```

Comment:
Is this a more accurate statement?

```suggestion
            // Ignore the first byte of the uncompressed public key
```

Reply:
Hmm its not really compressed/uncompressed, but a flag containing the compression instructions.

I think its better to put `compression` there: https://github.com/FuelLabs/fuel-crypto/pull/2/commits/b684ce163c059ccc0ca8af7dfb088316458edbc4